pkg/server/server.go

```diff
@@ -0,0 +1,184 @@
+package server
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/gorilla/mux"
+	"github.com/xenitab/azdo-proxy/pkg/auth"
+)
+
+type AzdoServer struct {
+	logger logr.Logger
+	port   string
+	proxy  *httputil.ReverseProxy
+	authz  auth.Authorization
+}
+
+func NewAzdoServer(logger logr.Logger, port string, authz auth.Authorization) *AzdoServer {
+	director := func(req *http.Request) {
```

Comment:
Can we make this a function / method outside of `NewAzdoServer`? Would make it easier to read.